### PR TITLE
First commit of actual code.

### PR DESCRIPTION
**Pull request by btisdall**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+RUN apk add --no-cache \
+  groff \
+  nodejs \
+  ;
+
+RUN pip install \
+  awscli \
+  ;
+
+RUN npm install -g \
```
Comment (ajohnstone) [inline Dockerfile]: Do you really need this?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+IMAGE=pbx-ecom-webs-awscli
```
Comment (ajohnstone) [inline Makefile]: If tied to quay.io this is redundant

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
 # pbx-ecom-webops-awscli
```
Comment (ajohnstone) [inline README.md]: Add quay.io badge

File: README.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+
 Docker image containing awscli and related AWS API command line tools.
+
+In particular this image includes [cfn-create-or-update](https://github.com/widdix/cfn-create-or-update) which, when working with CloudFormation, provides a comfortable way of creating or updating a stack without having to worry about whether the named stack already exists.
+
+The image is built and hosted by [quay.io](https://quay.io), making use of the the `Dockerfile` directly. There is a makefile for local image development, invoke as `make image`..
```
Comment (Gwenall-Photobox) [inline README.md]: there are 2 dots at the end of line

File: README.md [hunk 5]
```diff
@@ -1,2 +1,30 @@
...
+
+```
+docker run -v $HOME/.aws:/root/.aws -e 'AWS_PROFILE=photobox-dev' quay.io/photobox/pbx-ecom-webops-awscli ec2 describe-instances
+```
+
+If you want to run `cfn-create-or-update` then an invocation might be:
+
+```
+docker run -v $HOME/.aws:/root/.aws -e 'AWS_PROFILE=photobox-dev' --entrypoint cfn-create-or-update quay.io/photobox/pbx-ecom-webops-awscli --stack-name my-stack --template-body file://path/to/template.yaml --wait
+``
```
Comment (Gwenall-Photobox) [inline README.md]: it should be 3 back quotes at the end of line

File: README.md [hunk 6]
```diff
@@ -1,2 +1,30 @@
...
+
+## Run the container
+
+For human usage you will probably want to mount your AWS credentials directory when running the image; you will also need to set `AWS_PROFILE` unless you have a `default` profile and want to use that. Machines running the image will probably want to assume a role to obtain temporary credentials.
+
+The default `ENTRYPOINT` runs `awscli` with no arguments, so usage would be something like:
+
+
+```
+docker run -v $HOME/.aws:/root/.aws -e 'AWS_PROFILE=photobox-dev' quay.io/photobox/pbx-ecom-webops-awscli ec2 describe-instances
```
Comment (Gwenall-Photobox) [inline README.md]: Either a tag should be specified here or an image should be tagged `latest` in quay repository

File: Dockerfile (hunk 1 above)
Comment (btisdall) [inline Dockerfile]: Not sure if you mean do I need the tool or something else?

File: Makefile (hunk 2 above)
Comment (btisdall) [inline Makefile]: I explain in the `README` it's for local dev - do you think that's bad?

File: README.md (hunk 6 above)
Comment (ajohnstone) [inline README.md]: Do not use latest.

Comment (btisdall) [inline README.md]: Right!

File: Dockerfile (hunk 1 above)
Comment (ajohnstone) [inline Dockerfile]: I mean you add node + a module and all its dependencies that can be done in a single line to 3 lines of bash. Seems a unnecessary.

```
aws cloudformation update 
if [ $? -eq 255]; then
 ...
fi

```

File: Makefile (hunk 2 above)
Comment (ajohnstone) [inline Makefile]: Do you really need a makefile is what I am really asking?

```docker build -t xx .```

File: Dockerfile (hunk 1 above)
Comment (btisdall) [inline Dockerfile]: I disagree strongly. That code by itself handles an existing stack in a very simplistic way and would result in ugly output. Why write and maintain code when someone already has and all I need to do is package it once?

File: Makefile (hunk 2 above)
Comment (btisdall) [inline Makefile]: It's less typing so I'd say yes.

File: Dockerfile (hunk 1 above)
Comment (ajohnstone) [inline Dockerfile]: So why add node and not a python solution? Seems overkill to add node to a python base image.
+ any of the npm dependencies.

**Pull request merged**